src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -6680,20 +6693,29 @@ private StatementSyntax ParseStatementNoDeclaration(bool allowAnyExpression)
                     {
                         return this.ParseExpressionStatement(this.ParseQueryExpression(0));
                     }
-                    else
-                    {
-                        goto default;
-                    }
+                    goto default;
 
                 default:
-                    if (this.IsPossibleLocalDeclarationStatement(allowAnyExpression))
-                    {
-                        return null;
-                    }
-                    else
-                    {
-                        return this.ParseExpressionStatement();
-                    }
+                    return ParseExpressionStatementIfPossible(allowAnyExpression);
+            }
+        }
+
+        private SyntaxToken ParseAwaitKeywordForAsyncStreams()
+        {
+            Debug.Assert(this.CurrentToken.ContextualKind == SyntaxKind.AwaitKeyword);
+            SyntaxToken awaitToken = this.EatContextualToken(SyntaxKind.AwaitKeyword);
+            return CheckFeatureAvailability(awaitToken, MessageID.IDS_FeatureAsyncStreams);
+        }
+
+        private StatementSyntax ParseExpressionStatementIfPossible(bool allowAnyExpression)
```

Comment:
i would doc this method. Possibly call it "TryParse" to indicate it can return null.

Reply:
I've factoring this code back into `ParseStatementNoDeclaration`